### PR TITLE
TOOLS-2651: Replace build & test scripts with go-based build code

### PR DESCRIPTION
**Pull request by rychipman**
I've been poking at this on and off for a few weeks now, and curious to get some feedback on it. A lot of this is cribbed from ADL's build code.

### CODE REVIEW

File: common.yml [hunk 1]
```diff
@@ -386,73 +367,12 @@ functions:
         chmod +x ./bin/*
         rm -rf ./mongodb-*
 
-  "fetch tool" :
```
Comment (rychipman) [inline common.yml:389]: I removed the dependency that the qa tests have on dist. It's relatively cheap for us to build the tools in each task (only a few seconds in total), and it removes cruft from the evg project file

File: common.yml [hunk 2]
```diff
@@ -395,64 +376,12 @@ functions:
       local_file: src/github.com/mongodb/mongo-tools/bin/${tool}${extension}
       remote_file: mongo-tools/binaries/${build_id}/${edition|community}/${tool}${extension}
 
-  "generate coverage html + text":
```
[the remark this answers is not in the document]
Comment (tfogo) [reply common.yml:398]: Yeah I think once we do the js -> go test conversion we can start looking at coverage.

**Comment by tfogo** [comment]
This is awesome! I'm getting the following error:

```
env variable "EVG_VARIANT" not set
```

**Comment by rychipman** [comment]
Good catch! `EVG_VARIANT` is set on every evergreen distro, but I forgot to add code for detecting the platform outside of evergreen

**Comment by rychipman** [comment]
After this morning's discussion re: excluding build-system refactors from the Connectors Dependency Management scope, I think we can start moving forward on this again. I think the one open question that we may still want to discuss here was the mage vs. goke comparison.

My personal preference would be to stick with goke over mage, but would love to hear other thoughts as well. Some thoughts off the top of my head:
- A quick skim of mage's docs didn't turn up any clear differentiators of mage relative to goke
- The bootstrapping/build tag acrobatics that mage requires seem a bit unnecessary
- Given that goke is craig's project, we're more likely to be able to get new features merged if we need/want them!

Also adding @huan-Mongo as a reviewer so he can chime in on this item!

**Comment by huan-Mongo** [comment]
After reviewing the both Mage and Goke, I agree these two tools are very similar in terms of functionality. There are 3 places I see Mage has some edge.
1. Mage could build its dependencies both in parallel or sequentially, while Goke only supports sequential build. For example, in Mage, you can run the integration test and unit tests in parallel before doing release. 
2. Mage has no dependencies required for building and Goke has several.
3. Mage has some nice helper class like https://godoc.org/github.com/magefile/mage/sh which mainly helps to manipulate the output of command or make script easier to read. Though those classes are not complicated to be implemented by ourselves. 
 
I also agree none of these 3 points are indispensable to our use case.

**Review by tfogo — CHANGES_REQUESTED** [review]
LGTM... but I'd rather not call the build script `make`. I think that's confusing. I'd rather call it `build` or `goke` or something else.

**Pull request merged**